lock-api/src/main/java/com/palantir/lock/client/TransactionStarter.java

```diff
@@ -79,6 +80,18 @@ StartIdentifiedAtlasDbTransactionResponse startIdentifiedAtlasDbTransaction() {
         }
     }
 
+    List<Optional<StartIdentifiedAtlasDbTransactionResponse>> startIdentifiedAtlasDbTransactionBatch(int count) {
+        return autobatcher.applyBatch(IntStream.range(0, count).mapToObj($ -> (Void) null).collect(Collectors.toList()))
```

Comment:
I vote that this be implemented to return List<StartIdentifiedAtlasDbTransactionResponse> and throw Atlas's TransactionFailedRetriableException if not the whole batch is happy.

Reply:
The thing I was trying to avoid throughout is that you possibly increase the failure rate if you fail a whole batch based on possibly a single transaction's outcome; however, it may be that in general use we don't expect many transactions at all to fail to start, and therefore this is a reasonable assumption to make.